test/js/web/encoding/text-decoder.test.js

```diff
@@ -250,7 +250,7 @@ describe("TextDecoder", () => {
   it("constructor should set values", () => {
     const decoder = new TextDecoder("utf-8", { fatal: true, ignoreBOM: false });
     expect(decoder.fatal).toBe(true);
-    // expect(decoder.ignoreBOM).toBe(false); // currently the getter for ignoreBOM doesn't work and always returns undefined
+    expect(decoder.ignoreBOM).toBe(false);
```

Comment:
we haven't actually implemented BOM detection

Reply:
Oh, okay, let me try to implement the BOM detection in this PR, I will convert this to draft for now.